pkg/controller/replication/replication_controller.go

```diff
@@ -242,6 +220,7 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 	go rm.podController.Run(stopCh)
 	for i := 0; i < workers; i++ {
 		go wait.Until(rm.worker, time.Second, stopCh)
+		go wait.Until(rm.abandoner, 0, stopCh)
```

Comment:
We can probably have a map of workers, with a worker assigned to a given RC - that should help with most of the races.